File: src/com/fenghuo/adapter/NoteAdapter.java
package com.fenghuo.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.Button;
import android.widget.ImageView;

import com.fenghuo.LineTextView;
import com.fenghuo.db.DBNoteHelper;
import com.fenghuo.bean.Note;
import com.fenghuo.notes.R;

import java.util.List;

public class NoteAdapter extends BaseAdapter {

	private List<Note> data;
	private boolean isEdite;// 是否在编辑
	private LayoutInflater inflater;
	private Context context;

	public NoteAdapter(Context context, List<Note> list) {
		inflater = LayoutInflater.from(context);
		this.context = context;
		this.data = list;
	}

	public void Update(List<Note> list) {
		data.clear();
		data = list;
		notifyDataSetChanged();
	}

	/**
	 * 进入编辑状态
	 */
	public void startEdit() {
		isEdite = true;
		notifyDataSetChanged();
	}

	/**
	 * 退出编辑状态
	 */
	public void quitEdit() {
		isEdite = false;
		notifyDataSetChanged();
	}

	public boolean isEdit() {
		return isEdite;
	}

	@Override
	public int getCount() {

		return data.size();
	}

	@Override
	public Object getItem(int arg0) {

		return data.get(arg0);
	}

	@Override
	public long getItemId(int arg0) {

		return arg0;
	}

	@Override
	public View getView(int position, View view, ViewGroup arg2) {

		ViewHolder h = null;
		if (view != null) {
			h = (ViewHolder) view.getTag();
		} else {
			h = new ViewHolder();
			view = inflater.inflate(R.layout.item_note, null);
			h.item_img = (ImageView) view.findViewById(R.id.item_img);
			h.item_tv_content = (LineTextView) view
					.findViewById(R.id.item_tv_content);
			h.item_btn_delete = (Button) view.findViewById(R.id.item_btndelete);
			h.item_alarm = (ImageView) view.findViewById(R.id.item_alarm);
			view.setTag(h);
		}
		h.item_img.setBackgroundResource(data.get(position).getImg());
		h.item_tv_content.setText(data.get(position).getContent());
		h.item_btn_delete.setOnClickListener(new onclicklistener());
		h.item_btn_delete.setTag(position);
		// 此方法对横线的位置进行微调(原始:getPaddingtop+lineHeigt+x). 这里setLineDown(X)
		h.item_tv_content.setLineDown(-h.item_tv_content.getPaddingTop() - 1);
		if (data.get(position).getAlarm() == 1)
			h.item_alarm.setVisibility(View.VISIBLE);
		else
			h.item_alarm.setVisibility(View.INVISIBLE);
		if (isEdite)
			h.item_btn_delete.setVisibility(View.VISIBLE);
		else
			h.item_btn_delete.setVisibility(View.GONE);
		return view;
	}

	class onclicklistener implements View.OnClickListener {

		@Override
		public void onClick(View arg0) {

			int position = Integer.valueOf(arg0.getTag().toString());
			// 从列表中移除
			DBNoteHelper dbNoteHelper = new DBNoteHelper(context);
			dbNoteHelper.Delete(data.get(position).getId());
			dbNoteHelper.Desdroy();
			// 从集合中移除
			data.remove(position);
			notifyDataSetChanged();
		}

	}

	private class ViewHolder {
		public ImageView item_img;
		// public LineEditText item_tv_content;
		public LineTextView item_tv_content;
		public Button item_btn_delete;
		public ImageView item_alarm;
	}
}

File: src/com/fenghuo/notes/adapter/NoteAdapter.kt
package com.fenghuo.notes.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.Button
import android.widget.ImageView
import com.fenghuo.LineTextView
import com.fenghuo.notes.R
import com.fenghuo.notes.Values
import com.fenghuo.notes.bean.Note
import com.fenghuo.notes.db.DBNoteHelper

class NoteAdapter() : BaseAdapter(), View.OnClickListener {


    var isEdit: Boolean = false
    private var inflater: LayoutInflater? = null
    private var noteDbHelper: DBNoteHelper? = null
    private val data: MutableList<Note> = arrayListOf()

    constructor(context: Context) : this() {
        this.inflater = LayoutInflater.from(context)
        this.noteDbHelper = DBNoteHelper(context)
        this.data.clear()
    }

    /**
     * 刷新数据
     * **/
    fun Update() {
        data.clear()
        data.let { this.data.addAll(noteDbHelper!!.Getlist()) }
    }

    /**
     * 进入编辑状态
     */
    fun startEdit() {
        isEdit = true
        notifyDataSetChanged()
    }

    /**
     * 退出编辑状态
     */
    fun quitEdit() {
        isEdit = false
        notifyDataSetChanged()
    }

    fun destory() {
        noteDbHelper?.Desdroy()
        noteDbHelper = null
    }

    override fun getCount(): Int = data.size
    override fun getItem(arg0: Int): Any = data[arg0]
    override fun getItemId(arg0: Int): Long {
        return arg0.toLong()
    }

    override fun onClick(view: View) {
        val item: Note = view.getTag(R.id.tag_1) as Note
        // 从列表中移除
        val dbNoteHelper = DBNoteHelper(view.context)
        dbNoteHelper.Delete(item.id)
        dbNoteHelper.Desdroy()
        // 从集合中移除
        data.remove(item)
        notifyDataSetChanged()
    }

    override fun getView(position: Int, view: View?, arg2: ViewGroup): View {
        var view = view
        var h: ViewHolder? = null
        val item: Note = data[position]
        if (view != null) {
            h = view.tag as ViewHolder
        } else {
            h = ViewHolder()
            view = inflater?.inflate(R.layout.item_note, null)
            h.item_img = view!!.findViewById(R.id.item_img)
            h.item_tv_content = view.findViewById(R.id.item_tv_content)
            h.item_btn_delete = view.findViewById(R.id.item_btndelete)
            h.item_alarm = view.findViewById(R.id.item_alarm)
            view.tag = h
        }

        val imgIndex = item.img
        if (imgIndex >= 0 && imgIndex < Values.item_bg_preview.size) {
            h.item_img!!.setBackgroundResource(Values.item_bg_preview[imgIndex])
        } else {
            h.item_img!!.setBackgroundResource(R.drawable.page_blue)
        }

        h.item_tv_content!!.text = item.content
        h.item_btn_delete!!.setOnClickListener(this)
        h.item_btn_delete!!.setTag(R.id.tag_1, item)
        // 此方法对横线的位置进行微调(原始:getPaddingtop+lineHeigt+x). 这里setLineDown(X)
        h.item_tv_content!!.setLineDown(-h.item_tv_content!!.paddingTop - 1)
        if (item.alarm == 1)
            h.item_alarm!!.visibility = View.VISIBLE
        else
            h.item_alarm!!.visibility = View.INVISIBLE
        if (isEdit)
            h.item_btn_delete!!.visibility = View.VISIBLE
        else
            h.item_btn_delete!!.visibility = View.GONE
        view.setTag(R.id.tag_1, item)
        return view
    }

    private inner class ViewHolder {
        var item_img: ImageView? = null
        var item_tv_content: LineTextView? = null
        var item_btn_delete: Button? = null
        var item_alarm: ImageView? = null
    }

}
